### PR TITLE
ARROW-15572: [Java][Docs] Add Installation section to Java documentation

### PR DESCRIPTION
**Pull request by davisusanibar**
Jira ticket related: https://issues.apache.org/jira/browse/ARROW-15572

### CODE REVIEW

File: docs/source/java/install.rst [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+Configure your pom.xml with java module version needed. For example:
+
+.. code-block::
+
+    <dependency>
+        <groupId>org.apache.arrow</groupId>
+        <artifactId>arrow-vector</artifactId>
+        <version>7.0.0</version>
+    </dependency
```
Comment (lidavidm) [inline docs/source/java/install.rst:49]: There's a missing bracket here, also, shouldn't we show the 'full' set of dependencies needed? i.e. memory-netty, format, and vector? We also don't need to explain Maven or pom.xml here. Just state what modules are needed and show the example XML.

File: docs/source/java/install.rst [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    </settings>
+    $ mvn -Parrowrc clean install -X
+    Downloading from staged: https://repository.apache.org/content/repositories/staging/
+
+Installing from source
+----------------------
+
+See :ref:`java-development`.
+
+Installing Nightly Packages
```
Comment (lidavidm) [inline docs/source/java/install.rst:89]: If these are the nightly packages, then what is the purpose of the staging repository above?

File: docs/source/java/install.rst [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+Java Compatibility
+------------------
+
+Java modules is currently compatible with JDK 8 / 9 / 10 / 11.
+
+Using Maven
+-----------
+
+Downloading in Maven is triggered by a project declaring a dependency that is not present in the local repository.
```
[the remark this answers is not in the document]
Comment (davisusanibar) [reply docs/source/java/install.rst]: Deleted

File: docs/source/java/install.rst (hunk 1 above)
Comment (davisusanibar) [reply docs/source/java/install.rst]: Added libraries and xml demo

File: docs/source/java/install.rst [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+Installing Nightly Packages
+---------------------------
+
+.. warning::
+    These packages are not official releases. Use them at your own risk.
+
+All arrow nightly version are uploaded to github assets for example for March 01 it is uploading to `Github Nightly`_
+
+.. image:: img/java_nightly_github_asset.png
```
[the remark this answers is not in the document]
Comment (davisusanibar) [reply docs/source/java/install.rst]: Deleted

File: docs/source/java/install.rst (hunk 2 above)
Comment (davisusanibar) [reply docs/source/java/install.rst:113]: Nightly packages are added by arrow nightly jobs and uploaded to github assets.

For staging it is not aligned with github assets and it depend of dev/release/06-java-upload.sh.

If RC on nexus is closed, the artifacts will be available in the staging repository.

Comment (lidavidm) [reply docs/source/java/install.rst:113]: I still don't see why a developer should care about staging artifacts. Either they want the latest stable release or they want to test nightlies.

File: docs/source/java/install.rst [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+------------------
+
+Java modules are currently compatible with Java 8 / 9 / 10 / 11.
+
+Installing from Maven
+---------------------
+
+By default, Maven will download from the central repository: https://repo.maven.apache.org/maven2/org/apache/arrow/
+
+Configure your pom.xml with java module version needed, for example for: memory-netty, format, and vector
```
Comment (lidavidm) [inline docs/source/java/install.rst:38]: This is something that would be good to explain (though probably elsewhere in the docs): what is the difference between memory-netty and memory-unsafe

File: docs/source/java/install.rst (hunk 2 above)
Comment (davisusanibar) [reply docs/source/java/install.rst:77]: Clear, deleted staging maven section

**Review by lidavidm — APPROVED** [review]

File: docs/source/java/install.rst (hunk 5 above)
Comment (davisusanibar) [reply docs/source/java/install.rst:38]: Yes, this is part of this ticket https://issues.apache.org/jira/browse/ARROW-15573, working on that

**Comment by lidavidm** [comment]
@davisusanibar just a note for the future, try to make sure the PR title follows the expected format (`ARROW-12345: [Java][Doc] ...`)

File: docs/source/java/install.rst (hunk 5 above)
Comment (lidavidm) [inline docs/source/java/install.rst:38]:
```suggestion
Configure your pom.xml with the Java modules needed, for example:
``arrow-memory-netty``, ``arrow-format``, and ``arrow-vector``.
```

File: docs/source/java/install.rst [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+See :ref:`java-development`.
+
+Installing Nightly Packages
+---------------------------
+
+.. warning::
+    These packages are not official releases. Use them at your own risk.
+
+All arrow nightly version are uploaded to github assets for example for March 01 it is uploading to `Github Nightly`_
```
Comment (lidavidm) [inline docs/source/java/install.rst:83]:
```suggestion
Arrow nightly builds are uploaded to GitHub. For example, for 2022/03/01, they can be found at `Github Nightly`_.
```

File: docs/source/java/install.rst [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+For example if you need to test your code with these artifacts, then, you need to configure maven settings with:
```
Comment (lidavidm) [inline docs/source/java/install.rst:85]:
```suggestion
To test your code with these artifacts, then configure Maven with:
```

File: docs/source/java/install.rst [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+             </repository>
+          </repositories>
+          <id>arrownightly</id>
+        </profile>
+      </profiles>
+    </settings>
+    $ mvn -Parrownightly clean install -X
+    Downloading from staged: https://github.com/ursacomputing/crossbow/releases/tag/nightly-2022-03-01-0-github-java-jars/org/apache/arrow/arrow-vector/8.0.0.dev143/arrow-vector-8.0.0.dev143.pom
+
+Consider that you could see arrow nightly builds at `Arrow Nightly`_.
```
Comment (lidavidm) [inline docs/source/java/install.rst:115]:
```suggestion
Arrow nightly builds are posted on the mailing list at `builds@arrow.apache.org`_.
```

File: docs/source/java/install.rst [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+.. _Arrow Nightly: https://lists.apache.org/list.html?builds@arrow.apache.org
```
Comment (lidavidm) [inline docs/source/java/install.rst:117]:
```suggestion
.. _builds@arrow.apache.org: https://lists.apache.org/list.html?builds@arrow.apache.org
```

**Review by lidavidm — COMMENTED** [review]
Sorry, I just wanted to do a copy-editing pass.

**Review by lidavidm — APPROVED** [review]

**Pull request closed without merging**